src/app/component/user/auth/sign-up/sign-up.component.ts

```diff
@@ -55,7 +77,121 @@ export class SignUpComponent implements OnInit {
   private setNullAllMessage() {
     this.firstNameErrorMessageBackEnd = null;
     this.lastNameErrorMessageBackEnd = null;
+    this.userNameErrorMessageBackEnd = null;
     this.emailErrorMessageBackEnd = null;
     this.passwordErrorMessageBackEnd = null;
+    this.passwordConfirmErrorMessageBackEnd = null;
+  }
+
+  clickImg(elememtIdInput, image) {
+    const showEye = 'url(\'/assets/img/icon/eye.png\')';
+    const hideEye = 'url(\'/assets/img/icon/eye-show.png\')';
+    const passwordField = document.getElementById('password');
+    const imgEye = document.getElementById('img');
+    if (passwordField['type'] === 'password') {
+      imgEye.style.backgroundImage = hideEye;
+      passwordField.setAttribute('type', 'text');
+    } else {
+      imgEye.style.backgroundImage = showEye;
+      passwordField.setAttribute('type', 'password');
+    }
+  }
+
+  clickImgConfirm() {
+    const showEyeConfirm = 'url(\'/assets/img/icon/eye.png\')';
+    const hideEyeConfirm = 'url(\'/assets/img/icon/eye-show.png\')';
+    const confirmField = document.getElementById('password-confirm');
+    const imgEyeConfirm = document.getElementById('img-confirm');
+    if (confirmField['type'] === 'password') {
+      imgEyeConfirm.style.backgroundImage = hideEyeConfirm;
+      confirmField.setAttribute('type', 'text');
+    } else {
+      imgEyeConfirm.style.backgroundImage = showEyeConfirm;
+      confirmField.setAttribute('type', 'password');
+    }
+  }
+
+  matchPassword() {
+    const password = document.getElementById('password')['value'];
+    const confirmPassword = document.getElementById('password-confirm')['value'];
+    if (password !== confirmPassword) {
+      document.getElementById('seterror').style.display = 'block';
+      document.getElementById('password-confirm').style.border = '1px solid #F03127';
+    } else {
+      document.getElementById('seterror').style.display = 'none';
+      document.getElementById('password-confirm').style.border = '1px solid #839c94';
+    }
+  }
+
+  inputTextGreen() {
+    document.getElementById('first-name').style.color = '#13AA57';
+  }
+
+  inputTextBlack() {
```

Comment:
you can write mo generic method like 

inputBlack(fieldName) {
    document.getElementById(fieldName).style.color = '#000';
  }

Reply:
was added directive with the same functionality instead of previous code    https://github.com/ita-social-projects/GreenCityClient/pull/157